onnxruntime/test/global_thread_pools/test_inference.cc

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <core/common/make_unique.h>
+#include "core/session/onnxruntime_cxx_api.h"
+#include "core/graph/constants.h"
+#include "providers.h"
+#include <memory>
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <atomic>
+#include <gtest/gtest.h>
+#include "test_allocator.h"
+#include "../shared_lib/onnx_protobuf.h"
+#include "../shared_lib/test_fixture.h"
+#include <stdlib.h>
+
+struct Input {
+  const char* name = nullptr;
+  std::vector<int64_t> dims;
+  std::vector<float> values;
+};
+
+extern std::unique_ptr<Ort::Env> ort_env;
+static constexpr PATH_TYPE MODEL_URI = TSTR("testdata/squeezenet/model.onnx");
+class CApiTestGlobalThreadPoolsWithProvider : public testing::Test, public ::testing::WithParamInterface<int> {
+};
+
+template <typename OutT>
+static void RunSession(OrtAllocator* allocator, Ort::Session& session_object,
```

Comment:
>OrtAllocator* allocator [](start = 23, length = 23)

allocator should be a reference as it's required.

Reply:
Sure. This was copied from onnxruntime/test/shared_lib/test_inference.cc.